### PR TITLE
Improvements on Zk kerberos

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Now you can configure sasl gssapi on zookeeper independent of the kafka listeners
Should pave the way for sasl digest md5

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Created a new molecule scenario, that does zookeeper sasl protocol = kerberos and kafka just has scram, this validate the user creation not impacted

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.kafka_broker/templates/kafka_server_jaas.conf.j2 [hunk 1]
```diff
@@ -1,17 +1,19 @@
-// Specifies a unique keytab and principal name for each broker
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/templates/kafka_server_jaas.conf.j2:1]: this if statement is a bug, will need the kafkaserver block if kafka listeners have kerberos on them

question- is the KafkaServer block needed if just zookeeper sasl on?

File: roles/confluent.test/molecule/zookeeper-kerberos/verify.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
```
Comment (domenicbove) [inline roles/confluent.test/molecule/zookeeper-kerberos/verify.yml:1]: todo put in some tests

File: roles/confluent.zookeeper/tasks/main.yml [hunk 3]
```diff
@@ -36,7 +36,8 @@
     kerberos_user: "{{zookeeper_user}}"
     kerberos_keytab_path: "{{zookeeper_kerberos_keytab_path}}"
     kerberos_handler: "restart zookeeper"
-  when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms"
+  when: zookeeper_sasl_protocol == 'kerberos'
+  # when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms"
```
Comment (domenicbove) [inline roles/confluent.zookeeper/tasks/main.yml:40]: remove commented out code

**Comment by domenicbove** [comment]
Update the hosts_example w this variable

**Comment by JumaX** [comment]
@domenicbove Looks like ldap testing is failing for some reason:

```
00:06:41      TASK [confluent.test.ldap : Create certificate configuration] ******************
00:06:42  [31mfatal: [ldap1]: UNREACHABLE! => {"changed": false, "msg": "Failed to create temporary directory.In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg to a path rooted in \"/tmp\", for more error information use -vvv. Failed command was: ( umask 77 && mkdir -p \"` echo ~/.ansible/tmp `\"&& mkdir ~/.ansible/tmp/ansible-tmp-1588723602.68-8359-177795861671533 && echo ansible-tmp-1588723602.68-8359-177795861671533=\"` echo ~/.ansible/tmp/ansible-tmp-1588723602.68-8359-177795861671533 `\" ), exited with result 1", "unreachable": true}[0m
00:06:42      
00:06:42      RUNNING HANDLER [confluent.test.ldap : restart rsyslog] ************************
00:06:43  [31mfatal: [ldap1]: FAILED! => {"changed": false, "module_stderr": "Error: No such container: ldap1\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}[0m
00:06:43      
00:06:43      PLAY RECAP *********************************************************************
00:06:43      ldap1                      : ok=48   changed=35   unreachable=1    failed=1    skipped=2    rescued=0    ignored=0
00:06:43      zookeeper1                 : ok=7    changed=5    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 4]
```diff
@@ -115,7 +115,8 @@
     mode: 0640
     owner: "{{kafka_broker_user}}"
     group: "{{kafka_broker_group}}"
-  when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms"
+  # when: "'GSSAPI' in kafka_broker_sasl_enabled_mechanisms"
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:118]: remove commented out code

**Pull request merged**